### PR TITLE
Add Readme files for api and extension sample folders

### LINKED ISSUE #78: Add link to run-time mip-map generation tutorial
The api sample for generating mip-maps at run-time comes with a tutorial that hasn't been linked from the repo's readme.md.

It should be added to the tutorial list, and with it the list should maybe restructured, so api samples can be grouped together.

See [this discussion](https://github.com/KhronosGroup/Vulkan-Samples/pull/56/files/d67054282e2331169fade196da1c554472b6c83c).

### PR DESCRIPTION
**Pull request by SaschaWillems**
## Description

Adds dedicated readme files for the api and examples folders. The readme files contain a listing of the samples, along with a short description, and also links to tutorials (only one right now).

This also fixes #78.

## General Checklist:

Please ensure the following points are checked:

- n/a My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
-  n/a I have commented any added functions (in line with Doxygen)
-  n/a I have commented any code that could be hard to understand
-  n/a My changes do not add any new compiler warnings
-  n/a My changes do not add any new validation layer errors or warnings
-  n/a I have used existing framework/helper functions where possible
-  n/a My changes build and run on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)

### CODE REVIEW

File: samples/api/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+- limitations under the License.
+-
+-->
+
+# API samples <!-- omit in toc -->
+
+## Contents <!-- omit in toc -->
+
+- [Introduction](#introduction)
+- [Tutorials](#tutorials)
```
Comment (JoseEmilio-ARM) [inline samples/api/README.md:25]: Missing samples section

File: samples/extensions/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+- See the License for the specific language governing permissions and
+- limitations under the License.
+-
+-->
+
+# Extension samples <!-- omit in toc -->
+
+## Contents <!-- omit in toc -->
+
+- [Introduction](#introduction)
```
Comment (JoseEmilio-ARM) [inline samples/extensions/README.md:24]: shall we add samples here too?

File: samples/api/README.md (hunk 1 above)
Comment (SaschaWillems) [reply samples/api/README.md:25]: Just added it.

File: samples/extensions/README.md (hunk 2 above)
Comment (SaschaWillems) [reply samples/extensions/README.md:24]: Yes, makes sense here too. Just added it.

**Review by JoseEmilio-ARM — APPROVED** [review]

**Pull request merged**